packages/dht/src/connection/WebSocket/AutoCertifierClientFacade.ts

```diff
@@ -0,0 +1,77 @@
+import { AutoCertifierClient, SessionIdRequest, SessionIdResponse, CertifiedSubdomain, Certificate } from '@streamr/autocertifier-client'
+import { ListeningRpcCommunicator } from '../../exports'
+import { Logger, waitForEvent3 } from '@streamr/utils'
+
+const START_TIMEOUT = 60 * 1000
+
+const defaultAutoCertifierClientFactory = (
+    filePath: string,
+    autocertifierUrl: string,
+    autocertifierRpcCommunicator: ListeningRpcCommunicator,
+    wsServerPort: number
+) => new AutoCertifierClient(
+    filePath,
+    wsServerPort,
+    autocertifierUrl, (_, rpcMethodName, method) => {
+        autocertifierRpcCommunicator.registerRpcMethod(
+            SessionIdRequest,
+            SessionIdResponse,
+            rpcMethodName,
+            method
+        )                        
+    })
+
+export interface IAutoCertifierClient {
+    start(): Promise<void>
+    stop(): void
+    on(eventName: string, cb: (subdomain: CertifiedSubdomain) => void): void
+}
+
+interface AutoCertifierClientFacadeConfig {
+    autocertifierUrl: string
+    autocertifiedSubdomainFilePath: string
+    autocertifierRpcCommunicator: ListeningRpcCommunicator
+    wsServerPort: number
+    setHost: (host: string) => void
+    updateCertificate: (certificate: Certificate) => void
+    createClientFactory?: () => IAutoCertifierClient
```

Comment:
Maybe we could just pass in a client, as we create only on client anyway (and do it immediately in the constructor)

Reply:
Perhaps. This is implemnted as is to keep the `AutoCertifierClient` import inside the `AutoCertifierClientFacade` class